pandas-stubs/core/indexes/base.pyi

```diff
@@ -113,7 +107,7 @@ class Index(IndexOpsMixin, PandasObject):
         **kwargs,
     ) -> Self: ...
```

Comment:
Should probably add specific `__new__()` to return `DatetimeIndex`, `PeriodIndex`, `TimedeltaIndex`  .  In some sense, the pattern of `__new__()` here should mirror what is done for `Series`

Reply:
That would be needed when people use `pd.Index(<list of datetime>)`, or? If they use `pd.DatetimeIndex(...)` it will already work (but not check the type of the iterable)